Source/Platform/ExternalHttpProvider.cpp

```diff
@@ -0,0 +1,53 @@
+#include "pch.h"
+#include "ExternalHttpProvider.h"
+
+NAMESPACE_XBOX_HTTP_CLIENT_BEGIN
+
+ExternalHttpProvider& ExternalHttpProvider::Get() noexcept
+{
+    static ExternalHttpProvider s_instance;
+    return s_instance;
+}
+
+HRESULT ExternalHttpProvider::SetCallback(
+    HCCallPerformFunction performFunc,
+    void* context
+) noexcept
+{
+    RETURN_HR_IF(E_INVALIDARG, !performFunc);
+
+    m_perform = performFunc;
+    m_context = context;
+
+    return S_OK;
+}
+
+HRESULT ExternalHttpProvider::GetCallback(
+    HCCallPerformFunction* performFunc,
+    void** context
+) const noexcept
+{
+    RETURN_HR_IF(E_INVALIDARG, !performFunc);
+    RETURN_HR_IF(E_INVALIDARG, !context);
+
+    *performFunc = m_perform;
```

Comment:
there's a pretty big change in behavior from old libHC here, where you could get the "built in" perform function by calling `HCGetHttpCallPerformFunction` so that you could wrap the platform functionality in addition to wholesale replacement.

Reply:
that's a fair point, will have to think a bit about how we can maintain that option

Reply:
FWIW, it looks like this was bugged before too :)

The old implementation of HCGetHttpCallPerformFunction only retrieved user configured hooks, not the platform defaults.  I think we should be able to make this work like you (and the header docs) say it should

Reply:
FWIW I don't think this path wasn't actually possible before either, although I see that the docs suggest it is. HCGetHttpCallPerformFunction doesn't return the HCPerformEnv parameter that the default providers require, so they wouldn't be able to call them anyhow.

Reply:
Added an implementation that I believe works.  If you had something else in mind, let me know.